src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -375,6 +381,37 @@ public int compareOffsets(final NakadiCursor first, final NakadiCursor second) t
         return KafkaCursor.fromNakadiCursor(first).compareTo(KafkaCursor.fromNakadiCursor(second));
     }
 
+    public long totalEventsInPartition(final Timeline timeline, final String partitionString)
+            throws InvalidCursorOperation {
+        try {
+            final Timeline.StoragePosition positions = timeline.getLatestPosition();
+            if (positions == null) {
+                final Optional<PartitionStatistics> statsO = loadPartitionStatistics(timeline, partitionString);
```

Comment:
We don't expect too many calls to reach this branch. Actually, I'm afraid it might never get here. I just implemented this branch for the sake of completeness, in case this method would be used in another context in the future.